src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/CompilerServices/Conversions.vb

```diff
@@ -4,596 +4,1542 @@
+                        End If
+
+                    Case TypeCode.Int32
+                        If TypeOf Value Is Int32 Then
+                            Return CByte(DirectCast(Value, Int32))
+                        Else
+                            Return CByte(ValueInterface.ToInt32(Nothing))
+                        End If
+
+                    Case TypeCode.UInt32
+                        If TypeOf Value Is UInt32 Then
+                            Return CByte(DirectCast(Value, UInt32))
+                        Else
+                            Return CByte(ValueInterface.ToUInt32(Nothing))
+                        End If
+
+                    Case TypeCode.Int64
+                        If TypeOf Value Is Int64 Then
+                            Return CByte(DirectCast(Value, Int64))
+                        Else
+                            Return CByte(ValueInterface.ToInt64(Nothing))
+                        End If
+
+                    Case TypeCode.UInt64
+                        If TypeOf Value Is UInt64 Then
+                            Return CByte(DirectCast(Value, UInt64))
+                        Else
+                            Return CByte(ValueInterface.ToUInt64(Nothing))
+                        End If
+
+                    Case TypeCode.Decimal
+                        If TypeOf Value Is Decimal Then
+                            Return ValueInterface.ToByte(Nothing)
+                        Else
+                            Return CByte(ValueInterface.ToDecimal(Nothing))
+                        End If
+
+                    Case TypeCode.Single
+                        If TypeOf Value Is Single Then
+                            Return CByte(DirectCast(Value, Single))
+                        Else
+                            Return CByte(ValueInterface.ToSingle(Nothing))
+                        End If
+
+                    Case TypeCode.Double
+                        If TypeOf Value Is Double Then
+                            Return CByte(DirectCast(Value, Double))
+                        Else
+                            Return CByte(ValueInterface.ToDouble(Nothing))
+                        End If
+
+                    Case TypeCode.DateTime,
+                         TypeCode.Char
+                        ' Fall through to error
+
+                    Case TypeCode.String
+                        Dim StringValue As String = TryCast(Value, String)
+
+
+                        If StringValue IsNot Nothing Then
+                            Return CByte(StringValue)
+                        Else
+                            Return CByte(ValueInterface.ToString(Nothing))
+                        End If
+
+                    Case Else
+                        ' Fall through to error
+                End Select
+            End If
+
+            Throw New InvalidCastException(SR.Format(SR.InvalidCast_FromTo, VBFriendlyName(Value), "Byte"))
+
+        End Function
+
+        <CLSCompliant(False)>
```

Comment:
You can remove any CLSCompliant attributes. We don't use them.

Reply:
Hmm, I get various errors if I do this though:
```
"C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\Microsoft.VisualBasic.sln" (default target) (1) ->
"C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\tests\Microsoft.VisualBasic.Tests.csproj.metaproj" (d
efault target) (2) ->
"C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj.metaproj" (default t
arget) (3) ->
"C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj" (default target) (5
) ->
(CoreCompile target) ->
  C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft\VisualBasic\CompilerServices\Conversio
ns.vb(322,32): error BC40027: Return type of function 'ToSByte' is not CLS-compliant. [C:\Users\hughb\Documents\GitHub\
corefx\src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj]
  C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft\VisualBasic\CompilerServices\Conversio
ns.vb(343,32): error BC40027: Return type of function 'ToSByte' is not CLS-compliant. [C:\Users\hughb\Documents\GitHub\
corefx\src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj]
  C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft\VisualBasic\CompilerServices\Conversio
ns.vb(613,32): error BC40027: Return type of function 'ToUShort' is not CLS-compliant. [C:\Users\hughb\Documents\GitHub
\corefx\src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj]
  C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft\VisualBasic\CompilerServices\Conversio
ns.vb(634,32): error BC40027: Return type of function 'ToUShort' is not CLS-compliant. [C:\Users\hughb\Documents\GitHub
\corefx\src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj]
  C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft\VisualBasic\CompilerServices\Conversio
ns.vb(899,32): error BC40027: Return type of function 'ToUInteger' is not CLS-compliant. [C:\Users\hughb\Documents\GitH
ub\corefx\src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj]
  C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft\VisualBasic\CompilerServices\Conversio
ns.vb(920,32): error BC40027: Return type of function 'ToUInteger' is not CLS-compliant. [C:\Users\hughb\Documents\GitH
ub\corefx\src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj]
  C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft\VisualBasic\CompilerServices\Conversio
ns.vb(1189,32): error BC40027: Return type of function 'ToULong' is not CLS-compliant. [C:\Users\hughb\Documents\GitHub
\corefx\src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj]
  C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft\VisualBasic\CompilerServices\Conversio
ns.vb(1212,32): error BC40027: Return type of function 'ToULong' is not CLS-compliant. [C:\Users\hughb\Documents\GitHub
\corefx\src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj]
  C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft\VisualBasic\CompilerServices\Conversio
ns.vb(2201,55): error BC40028: Type of parameter 'Value' is not CLS-compliant. [C:\Users\hughb\Documents\GitHub\corefx\
src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj]
  C:\Users\hughb\Documents\GitHub\corefx\src\Microsoft.VisualBasic\src\Microsoft\VisualBasic\CompilerServices\Conversio
ns.vb(2209,55): error BC40028: Type of parameter 'Value' is not CLS-compliant. [C:\Users\hughb\Documents\GitHub\corefx\
src\Microsoft.VisualBasic\src\Microsoft.VisualBasic.vbproj]

    0 Warning(s)
    10 Error(s)
```

Reply:
Hmm OK fair enough